bin/merge_sces.R

```diff
@@ -224,25 +185,94 @@ sce_list <- sce_list |>
     }
     sce$additional_modalities <- additional_modalities
 
-    if ("adt" %in% additional_modalities) {
-      retain_coldata_columns <- c(retain_coldata_columns, "adt_scpca_filter")
-    }
-
     metadata(sce)$cluster_algorithm <- NULL
     metadata(sce)$cluster_weighting <- NULL
     metadata(sce)$cluster_nn <- NULL
 
     return(sce)
   })
 
+# Determine SCE columns to retain  ---------------------------------------------
+
+# Define all possible colData columns which we might want to retain
+possible_coldata <- c(
+  "barcodes",
+  # RNA statistics and post-processing
+  "sum",
+  "detected",
+  "total",
+  "subsets_mito_sum",
+  "subsets_mito_detected",
+  "subsets_mito_percent",
+  "miQC_pass",
+  "prob_compromised",
+  "scpca_filter",
+  # cell type names
+  "submitter_celltype_annotation",
+  "singler_celltype_annotation",
+  "singler_celltype_ontology",
+  "cellassign_celltype_annotation",
+  "cellassign_max_prediction",
+  # ADT statistics
+  "adt_scpca_filter",
+  "altexps_adt_sum",
+  "altexps_adt_detected",
+  "altexps_adt_percent",
+  # cellhash statistics & demux results
+  "altexps_cellhash_sum",
+  "altexps_cellhash_detected",
+  "altexps_cellhash_percent",
+  "hashedDrops_sampleID",
+  "HTODemux_sampleID",
+  "vireo_sampleid"
+)
+
+# Define colData columns to retain based on intersection with present columns
+retain_coldata_columns <- sce_list |>
+  purrr::map(\(sce) names(colData(sce))) |>
+  purrr::reduce(union) |>
+  intersect(possible_coldata)
+
+# Define altExp columns to retain/preserve, currently only for "adt"
+adt_possible_coldata <- c(
+  "zero.ambient",
+  "high.ambient",
+  "ambient.scale",
+  "sum.controls",
+  "high.controls",
+  "discard"
+)
+adt_possible_columns <- sce_list |>
```

Comment:
Since this is now what is there, not just what could be, suggesting a rename.
```suggestion
adt_columns_present <- sce_list |>
```

Reply:
this was my bad..!